eth_client/lib/eth_client/application.ex

```diff
@@ -10,15 +10,17 @@ defmodule EthClient.Application do
 
   @impl true
   def start(_type, _args) do
+    {chain_id, _} = Integer.parse(System.get_env("ETH_CHAIN_ID", "1234"))
```

Comment:
You can extract this code as a function:

```elixir
def get_chain_id() do
   {chain_id, _} =
      "ETH_CHAIN_ID"
      |> System.get_env("1234")
      |> Integer.parse()
      
    chain_id
end
```
It looks like more code, however, it is more readable and you can reuse it elsewhere with one line only:

```elixir
MyModule.get_chain_id()
```
Another advantage that seems unimportant but has a significant impact is that if you need to rename the environment variable you only affect a single module, if you invoke the variable everywhere you would have to modify it in each place where you use it which increases the margin of human error.